VPC-Section-Formatting (#141)
C2Devel/terraform-provider-croccloud: docs/data-sources/subnet.md

Reviewer: а вот здесь кстати мн.ч. -- available sunets, может быть и выше стоит поменять число
@PixarV  а почему в регионе? мы ищем по всем проектам?

Reviewer 2: Тут имеется ввиду в регионе в рамках конкретного проекта. Проект указывается при указании access id.

Кажется, что много где встречалось упоминание региона без упоминания проекта. Думаю, что проект подразумевается, если не оговорено, что сущность глобальная.

Provides information about a subnet.

> It can be used when a module accepts a subnet ID as an input variable and needs to, for example, determine the ID of the VPC that the subnet belongs to.

Reviewer: ```suggestion
The data source can be used when a module accepts a subnet ID as an input variable and needs to, for example, determine the ID of the VPC that the subnet belongs to.
```

## Example Usage

### Specific Examples

> The following example shows how one might accept a subnet ID as a variable and use this data source to obtain the data necessary to create a security group that allows connections from hosts in that subnet.

Reviewer: выше было исправлено вроде так
```suggestion
The following example shows how to accept a subnet ID as a variable and use this data source to obtain the data necessary to create a security group that allows connections from hosts in that subnet.
```

}
```

## Argument reference

The arguments of this data source act as filters for querying the available subnets in the current project within the current region.
The given filters must exactly match one subnet whose data will be exported as attributes.

The following arguments are optional:

* `availability_zone` - (Optional, Integer) The availability zone where the subnet must reside.
* `default_for_az` - (Optional, Boolean) Indicates whether the desired subnet must be the default subnet for its associated availability zone.
    * _Default value_: `true`

>     ~> **Note** Setting this argument to `false` does not actually filter by it *not* being the default, because Terraform can't distinguish between `false` and "not set".

Reviewer: в других местах cannot

* `filter` - (Optional, Block) One or more name/value pairs to use as filters.
    * _Valid values:_ See supported names and values in [EC2 API documentation][describe-subnets]
* `id` - (Optional, String) The ID of the specific subnet to retrieve.
* `state` - (Optional, String) The state that the desired subnet must have.
* `tags` - (Optional, Map of strings) Key-value pairs. Must exactly match pairs on the required resource.
* `vpc_id` - (Optional, String) The ID of the VPC that the desired subnet belongs to.

## Attribute reference

### Supported attributes

> In addition to all arguments above, the following attributes are exported:

Reviewer: по идее здесь надо в ед.ч., но есть и другие места, где мн.ч. вместо ед. используется

}
```

## Argument reference

The arguments of this data source act as filters for querying the available subnets in the current project within the current region.

~> **Note** The given filters must exactly match the resource whose data will be exported as attributes.

The following arguments are optional:

> * `availability_zone` - (Optional, Integer) The availability zone where the subnet must reside.

Reviewer 2: ```suggestion
* `availability_zone` - (Optional, String) The availability zone where the subnet must reside.
```

* `default_for_az` - (Optional, Boolean) Indicates whether the desired subnet must be the default subnet for its associated availability zone.

>     * _Default value_: `true`

Reviewer 2: ```suggestion
    * _Default value_: `false`
```

    ~> **Note** Setting this argument to `false` does not actually filter by it *not* being the default, because Terraform cannot distinguish between `false` and "not set".

* `filter` - (Optional, [Block](#filter)) One or more name/value pairs to use as filters.
    * _Valid values:_ See supported names and values in [EC2 API documentation][describe-subnets]
* `id` - (Optional, String) The ID of the specific subnet to retrieve.

> * `state` - (Optional, String) The state that the desired subnet must have.

Reviewer 2: Перемести в атрибуты пожалуйста

* `tags` - (Optional, Map of strings) Key-value pairs. Must exactly match pairs on the required resource.
* `vpc_id` - (Optional, String) The ID of the VPC that owns the desired subnet.

### filter

* `name` - (Required, String) The name of the filter.
    * _Constraints:_ Filter names are case-sensitive
* `values` - (Required, List of strings) One or more filter values.
    * _Constraints:_ Filter values are case-sensitive

## Attribute reference

> ### Supported attributes

Reviewer 2: Добавь в список поддерживаемых атрибутов еще пожалуйста `available_ip_address_count`, `cidr_block`

Author: Судя по коду, `cidr_block` - это аргумент, а не атрибут. Добавил всё же в аргументы, проверь правильность плз)
```
			"cidr_block": {
				Type:     schema.TypeString,
				Optional: true,
				Computed: true,
			}
```

Reviewer 2: В качестве аргумента мы его не поддерживаем, а в качестве атрибута поддерживаем (заполняем корректным значением). Так что перенеси в атрибуты пожалуйста

In addition to all arguments above, the following attribute is exported:

* `map_public_ip_on_launch` - (Boolean) Indicates whether public IP addresses will be associated with instances created in this subnet. Addresses are associated only if there are available allocated Elastic IP addresses.

> ### Unsupported attributes

Reviewer 2: Добавь в список неподдерживаемых атрибутов еще пожалуйста `ipv6_cidr_block`

* `filter` - (Optional, [Block](#filter)) One or more name/value pairs to use as filters.
    * _Valid values:_ See supported names and values in [EC2 API documentation][describe-subnets]
* `tags` - (Optional, Map of strings) Key-value pairs. Must exactly match pairs on the required resource.
* `vpc_id` - (Optional, String) The ID of the VPC that owns the desired subnet.

### filter

* `name` - (Required, String) The name of the filter.
    * _Constraints:_ Filter names are case-sensitive
* `values` - (Required, List of strings) One or more filter values.
    * _Constraints:_ Filter values are case-sensitive

## Attribute reference

### Supported attributes

In addition to all arguments above, the following attribute is exported:

* `available_ip_address_count` - (Integer) The number of remaining available IPv4 addresses in the subnet.
* `cidr_block` - (String) The IPv4 CIDR block for the subnet.
* `map_public_ip_on_launch` - (Boolean) Indicates whether public IP addresses will be associated with instances created in this subnet. Addresses are associated only if there are available allocated Elastic IP addresses.
* `state` - (String) The state that the desired subnet must have.

> available_ip_address_count, cidr_block

Reviewer 2: что-то лишнее

## Argument Reference

> The arguments of this data source act as filters for querying the available subnets in the current project within the current region. The given filters must match exactly one subnet whose data will be exported as attributes.

Reviewer: available

The following arguments are optional:

* `availability_zone` - (Optional, Integer) The availability zone where the subnet must reside.

> * `default_for_az` - (Optional, Boolean) Indicated whether the desired subnet must be the default subnet for its associated availability zone.

Reviewer: ```suggestion
* `default_for_az` - (Optional, Boolean) Indicates whether the desired subnet must be the default subnet for its associated availability zone.
```

    * _Default value_: `true`

>     ~> **Note** Setting this argument to "false" does not actually filter by it *not* being the default, because Terraform can't distinguish between "false" and "not set".

Reviewer: ```suggestion
    ~> **Note** Setting this argument to `false` does not actually filter by it *not* being the default, because Terraform can't distinguish between `false` and "not set".
```

The following arguments are optional:

> * `availability_zone` - (Optional) (Int) Availability zone where the subnet must reside.

Reviewer: The availability zone

> * `default_for_az` - (Optional) (Bool) Whether the desired subnet must be the default subnet for its associated availability zone.

Reviewer: Indicates whether

>   * _Default value_: `true`.

Reviewer: без точки

> ~> **Note** Setting this argument to "false" does not actually filter by it *not* being the default, because Terraform can't distinguish between "false" and "not set".

Reviewer: note разбивает список. Я бы отступы сделал, чтобы он относился к конкретному пункту
`false`

* `filter` - (Optional) One or more name/value pairs to use as filters.
    * _Valid values:_ See supported names and values in [EC2 API documentation][describe-subnets]
* `id` - (Optional) (String) The ID of the specific subnet to retrieve.

> * `state` - (Optional) (String) State that the desired subnet must have.

Reviewer: The state
